From 3f794c87053067103a3d20d44057adb31f1a51fa Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Mon, 3 Sep 2012 15:07:32 +0200
Subject: [PATCH 127/132] tmpfiles: allow Age to be set to 0

Mostly useful for testing purposes. Setting Age to 1s works just as
well, but it is surprising that using 0s (or just 0) does not work.

Also clarify this in the documentation.
(cherry picked from commit a2558205f762ce6f1374cea0eedf5e76a6134f41)
---
 man/tmpfiles.d.xml      | 7 ++++++-
 src/tmpfiles/tmpfiles.c | 2 +-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/man/tmpfiles.d.xml b/man/tmpfiles.d.xml
index a86ef33..86fa439 100644
--- a/man/tmpfiles.d.xml
+++ b/man/tmpfiles.d.xml
@@ -265,7 +265,12 @@ L    /tmp/foobar -    -    -    -   /dev/null</programlisting>
                         </variablelist>
 
                         <para>If multiple integers and units are specified the time
-                        values are summed up.</para>
+                        values are summed up. If an integer is given witohut a unit,
+                        s is assumed.
+                        </para>
+
+                        <para>When the age is set to zero, the files are cleaned
+                        unconditionally.</para>
 
                         <para>The age field only applies to lines starting with
                         d, D and x. If omitted or set to - no automatic clean-up
diff --git a/src/tmpfiles/tmpfiles.c b/src/tmpfiles/tmpfiles.c
index bf900fa..0138c63 100644
--- a/src/tmpfiles/tmpfiles.c
+++ b/src/tmpfiles/tmpfiles.c
@@ -409,7 +409,7 @@ static int clean_item(Item *i) {
             i->type != IGNORE_PATH)
                 return 0;
 
-        if (!i->age_set || i->age <= 0)
+        if (!i->age_set)
                 return 0;
 
         n = now(CLOCK_REALTIME);
-- 
1.8.1

